app/pages/browser/onboardingMethodPage/onboardingMethodPage.ts

```diff
@@ -16,28 +16,31 @@ export class OnboardingMethodPage implements OnInit {
 
   // ctor 
   constructor(private nav: NavController,
-              private navParams: NavParams,
-              private onboardingDataService: OnboardingDataService) {
+    private navParams: NavParams,
+    private onboardingDataService: OnboardingDataService) {
     this.name = navParams.get("name");
   }
 
   // handles page init.
   ngOnInit() {
     this.onboardingDataService.initiateGetOnboardingMethod(this.name)
-            .then((files) => {
-                this.items = files;
-            }).catch((err) => {
-                // there was an error. display it on screen.
-                console.log(JSON.stringify(err));
-                doAlert(JSON.stringify(err));
-            });
+      .then((files) => {
+        this.items = files;
+      }).catch((err) => {
+        // there was an error. display it on screen.
+        console.log(JSON.stringify(err));
+        doAlert(JSON.stringify(err));
+      });
   }
 
+  // browses to an item
   browseItem(item: string) {
-    if (item.endsWith(".xml")) // it"s the XML file
-      this.nav.push(OnboardingFileContentPage, {path: this.name + "/" + item, name: item});
-    else
-      this.nav.push(OnboardingMethodNodePage, {method: this.name, name: item});
+    if (item.endsWith(".xml")) {
+      // if user tapped on the XML file, navigate to the contents
+      this.nav.push(OnboardingFileContentPage, { path: this.name + "/" + item, name: item });
```

Comment:
It may be possible to do the conditional statement (if item ends with .xml...) in the navPush directive, but even if it was that would (in my opinion) be too much complicated logic in the view (html). My goal with this commit was to clean up the simple one-line navigations to use navPush. For complicated conditional stuff like this, I think the current model is fine.

BTW this line of code needs to be cleaned up anyway in the future to be more robust. What if there's another file in this directory, not xml (e.g. a json schema file). We should just detect that it's a file rather than a folder and navigate to the contents if so. That should happen later.